---
title: "Start here"
author: "Hyunwoo Cho"
date: "`r Sys.Date()`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(
  warning=F,
  message=F,
  fig.width=7, fig.height=7,
  echo=T
)
```

## Preparing the data

### Download the sequence data from SRA and convert to `FASTQ` using this script.

```{bash download1, eval=F}
prefetch SRR111111xx
fastq-dump SRR111111xx.sra --split-files --gzip -O SRR111111xx/
```

The `MD5` sum of the `fastq` files is attached below.

```{bash md5sum-raw, eval=F}
md5sum *.fastq.gz
fff...
```

### Run alignment using this reproducible script:

```{bash align1, eval=F}
wget -c https://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_38/GRCh37_mapping/gencode.v38lift37.annotation.gtf.gz
gunzip -c gencode.v38lift37.annotation.gtf.gz

STAR --version
2.5.0a
STAR --genomeDir {how-was-this-made} --readFilesIn {L1_R1},{L2_R1} {L1_R2},{L2_R2} --runThreadN 12 --outFileNamePrefix results/results_ \
  --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --outSAMattributes All --outSAMunmapped Within --readFilesCommand zcat

htseq-count --version
0.13.5
htseq-count -f bam -r pos -s no -t CDS -i gene_name -m union Proj_11351_C_s_E223D0P531WT1.bam gencode.v38lift37.annotation.gtf > Proj_11351_C_s_E223D0P531WT1-htseq-count-matrix.txt
```

The output of the gene quantification can also be accessed at <https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE18xxxx>.

The `MD5` sum of the gene quantification files is attached below.

```{bash md5sum-processed, eval=F}
md5sum *-htseq-count-matrix.txt
79fa5a164ee02c16c59bef06697a9d1c  Proj_11351_C_s_E223D0P531WT1-htseq-count-matrix.txt
cc4d5f75f9625560a2e575cda67249e2  Proj_11351_C_s_E223D0P531WT2-htseq-count-matrix.txt
049542fc33223c02bc6d095274f66f25  Proj_11351_C_s_E223D1CULTUREP53WT1-htseq-count-matrix.txt
96bc008e91c147f2747977a4e82a5401  Proj_11351_C_s_E223D1CULTUREP53WT2-htseq-count-matrix.txt
d24d8c218f8f2d486ca771d5df4e4198  Proj_11351_C_s_E223D1GRAFTP53WT1-htseq-count-matrix.txt
0b8ab35617d9cb5ddea79e6e37e2acda  Proj_11351_C_s_E223D1GRAFTP53WT2-htseq-count-matrix.txt
```

```{r load-1, include=F}
library(data.table); library(rtracklayer); library(DESeq2); library(magrittr)
```

Load the gene quantification results and make a `DESeq2` object.

```{r prepare-1}
library(data.table); library(rtracklayer); library(DESeq2); library(magrittr)

fn = dir(path=".", pattern="*-htseq-count-matrix.txt", recursive=T, full.names=T)
fn %>% file.exists

mySampleName = gsub("Proj_11351_C_s_E223", "", gsub("-htseq.*", "", basename(fn)))
mySampleTable = data.frame(sampleName=mySampleName, fileName=fn, condition=factor(c("D0", "D0", "D1Culture", "D1Culture", "D1Graft", "D1Graft"), levels=c("D0", "D1Culture", "D1Graft")))

DESeq2Obj = DESeqDataSetFromHTSeqCount(sampleTable=mySampleTable, directory=".", design= ~ condition)
DESeq2Obj %<>% DESeq
DESeq2Obj.vst = varianceStabilizingTransformation(DESeq2Obj)
dim(DESeq2Obj)
```

There goes the PCA plot (Fig. 3).

```{r pca-1}
p1 = plotPCA(DESeq2Obj.vst, ntop=3000, intgroup="condition", returnData=T)
# Quick PCA plot
library(ggplot2); library(ggrepel)
percentVar = round(100 * attr(p1, "percentVar"))
ggplot() + geom_point(data=p1, aes(PC1, PC2, color=condition), size=4.0) +
  xlab(paste0("PC1: ", percentVar[1], "%")) + ylab(paste0("PC2: ", percentVar[2], "%")) + coord_fixed() +
  theme(panel.grid=element_blank()) +
  geom_text_repel(data=p1, aes(PC1, PC2, label=row.names(p1)), size=3, max.overlaps=30, segment.alpha=0.1, box.padding=0.2, point.padding=0.20)
```

The accompanying dendrogram (Fig. S3) goes here.

```{r pca-2, fig.show="hold"}
ntop.3000.selector = rowVars(assay(DESeq2Obj.vst)) %>% order(decreasing=T) %>% head(3000)
library(NMF); library(amap)
row.hclust = hcluster(dist(t(assay(DESeq2Obj.vst)[ntop.3000.selector, ])), method="euclidean", link="single")
row.hclust.d = as.dendrogram(row.hclust)
labels(row.hclust.d)

# normalized DESeq2 counts, log2 transform, top 3000 genes, correlation metric, ward linkage
row.hclust.all = hcluster(dist(t(log2(counts(DESeq2Obj, normalized=T) + 1)[ntop.3000.selector, ])), method="correlation", link="ward")
library(dendextend)
row.hclust.all.d = as.dendrogram(row.hclust.all)
row.hclust.all.d = row.hclust.all.d %>% set("labels", c("Day 1 Culture replicate 1", "Day 1 Culture replicate 2", "Day 0 replicate 1", "Day 0 replicate 2", "Day 1 Graft replicate 1", "Day 1 Graft replicate 2")) %>% set("leaves_pch", 19) %>% set("leaves_cex", 1.5) %>% set("leaves_col", c("green", "green", "red", "red", "blue", "blue"))
par(mar=c(0.6, 0.6, 0.6, 10.1))
plot(row.hclust.all.d, sub="", xlab="", yaxt="n", horiz=T)
```

```{r pc-coordinates-1}
p1
```

Please check that the dendrogram is topologically equivalent to one used in the figure. We made manual edits on it.






======











This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:


```{r cars}
summary(cars)
```

## Including Plots

You can also embed plots, for example:

```{r pressure, echo=FALSE}
plot(pressure)
```

Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.
